packages/rest/src/__tests__/resource-endpoint.ts

```diff
@@ -39,17 +39,19 @@ describe('Resource', () => {
   });
 
   it('should init', () => {
+    const author = UserResource.fromJS({ id: 5 });
     const resource = CoolerArticleResource.fromJS({
       id: 5,
       title: 'happy',
-      author: 5,
+      author,
     });
     expect(resource.pk()).toBe('5');
     expect(CoolerArticleResource.pk(resource)).toBe('5');
     expect(resource.title).toBe('happy');
     expect(resource.things).toBe('happy five');
     expect(resource.url).toBe('http://test.com/article-cooler/5');
-    expect(resource.author).toBe(5);
+    expect(resource.author).toBe(author);
+    expect(resource.author?.pk()).toBe('5');
```

Comment:
why the optional chain?

Reply:
In this particular case, user is not always defined as defined `readonly author: UserResource | null = null;`  in ` __tests__/new.ts `

Nested members always existing is also supported.

Reply:
ah, the previous assertion doesn’t type narrow, got it

Reply:
ya for some reason expect() doesn't integrate with typescript. if it threw, i think since 3.7 of typescript it would actually help, but it's not marked that way